File: GFG/Queue/llimp.cpp
#include<iostream>
#include"../ds.h"
using namespace std;
//for integer class datatype
class mQue{
    public:
    mQue() : sz(0) {}
    LinkedList<int> ll;
    int sz;
    void enqu(int value){
        ll.insert(value);
        sz++;
        return;
    }
    int dequ(){
        Node<int>* res=ll.head;
        int ret=res->data;
        ll.head=ll.head->next;
        delete res;
        sz--;
        return ret;
    }
    bool isEmp(){
        return sz==0;
    }
    int size(){
        return sz;
    }
    int front(){
        return ll.head->data;
    }
    int rear(){
        return ll.tail->data;
    }
};
int main(){
    mQue q;
    for(int i=1;i<11;i++){
        q.enqu(i*10);
    }
    cout<<q.size()<<" "<<q.isEmp()<<endl;
    while(q.isEmp()==false){
        cout<<q.rear()<<" "<<q.front()<<endl;
        cout<<q.dequ()<<endl;
    }
    return 0;
}

File: GFG/Queue/llimp.py
import ds

class mQue:
    def __init__(self):
        self.ll=ds.LinkedList()
        self.sz=0
        self.tail=None
    
    def size(self):
        return self.sz
    def isEmp(self):
        return self.sz==0
    def enqu(self,data):
        if(self.isEmp()):
            self.ll.insert(data)
            self.tail=self.ll.head
        else:
            self.ll.insert(data)
            self.tail=self.tail.next
        self.sz+=1
        return
    def dequ(self):
        res=self.ll.head.data
        self.ll.head=self.ll.head.next
        self.sz-=1
        return res
    def front(self):
        return self.ll.head.data
    def rear(self):
        return self.tail.data
    
q=mQue()
for i in range(1,11):
    q.enqu(i*10)
print(q.size()," ",q.isEmp())
while(q.isEmp()==False):
    print(q.front()," ",q.rear())
    print(q.dequ())